apis/tempo/v1alpha1/config.go

```diff
@@ -0,0 +1,58 @@
+package v1alpha1
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// Values represent parts of the config.
+type Values map[string]interface{}
+
+// DeepCopy make a deep copy of the Values structure.
+func (v *Values) DeepCopy() *Values {
+	out := make(Values, len(*v))
+	for key, val := range *v {
+		switch val := val.(type) {
+		case string:
+			out[key] = val
+
+		case []string:
+			out[key] = append([]string(nil), val...)
+		default:
+			out[key] = val
+		}
+	}
+	return &out
+}
+
+// Config encapsulates arbitrary config.
+type Config struct {
```

Comment:
can we have some unit tests for this?

Reply:
I was wondering on accept the @andreasgerstmayr suggestion and see if we can replace it.